src/ring_theory/polynomial/vieta.lean

```diff
@@ -68,6 +68,13 @@ begin
     exact nat.sub_lt_succ s.card k }
 end
 
+lemma _root_.finset.prod_X_sub_C_coeff {σ} (s : finset σ) (r : σ → R) (k : ℕ) (h : k ≤ s.card) :
+  (∏ i in s, (X + C (r i))).coeff k = ∑ t in s.powerset_len (s.card - k), ∏ i in t, r i :=
+begin
+  rw [finset.prod, ← map_map (λ r, X + C r) r, prod_X_add_C_coeff, finset.esymm_map_val];
+  rw s.val.card_map r, exacts [rfl, h],
+end
```

Comment:
I think it might make sense to extract the goal
```
(map (λ i, X + C (r i)) s.val).prod.coeff k = (map r s.val).esymm (s.card - k)
```
as a standalone lemma here; it feels a bit like how we state many continuity lemmas with an extra function (`r` here) so that the unifier can deal with composition for us.

Reply:
Done! Also fixed the wrong name (sub -> add).